Control-User-Interfaces/writing-sal-scripts.rst

```diff
@@ -0,0 +1,948 @@
+      obs_filter:
+        type: array
+        minItems: 1
+        items:
+          type: string
+        description: List of filter names for each exposure.
+      obs_grating:
+        type: array
+        minItems: 1
+        items:
+          type: string
+        description: List of grating names for each exposure.
+  """
+
+  return yaml.safe_load(schema)
+
+.. _Writing-SAL-Scripts-Handling-Configuration:
+
+Handling Configuration
+----------------------
+
+Having the configuration schema defined we now must implement how the SAL Script handle the input configuration.
+This operation is handled by the ``configure`` method.
+The method receives the configuration as a ``type.SimpleNamespace`` Python object.
+
+In the configuration schema written above, we noted that the schema is not verifying that the sizes of the input arrays (e.g. ``exptime``, ``obs_filter`` and ``obs_grating``) are consistent.
+That means we need to do it in the ``configure`` method.
+
+With all that in consideration the ``configure`` method would look like this:
+
+.. code-block:: python
+
+  async def configure(self, config):
+    """Handle script input configuration.
+
+    Parameters
+    ----------
+    config: `types.SimpleNamespace`
+      Configuration data. See `get_schema` for information about data
+      structure.
+
+    Raises
+    ------
+    RuntimeError:
+      If sizes of the configuration arrays (exptime, obs_filter and
+      obs_grating) are different.
+
+    """
+
+    # Check that exptime, obs_filter and obs_grating have the same time.
+    # Raise RuntimeError exception if not.
+
+    if not (len(config.exptime) == len(config.obs_filter) == len(obs_grating)):
+        raise RuntimeError(
+            f"Inconsistent size of arrays: exptime: {len(config.exptime)}, "
+            "obs_filter: {len(config.obs_filter)}, obs_grating: {len(obs_grating)}."
+        )
+
+    # We can also log information about the configuration
+
+    self.log.debug(
+        f"target_name: {config.target_name}, rot_value: {config.rot_value}, "
+        f"rot_type: {config.rot_type}, exptime: {config.exptime}, "
+        f"obs_filter: {config.obs_filter}, obs_grating: {config.obs_grating}."
+    )
+
+    # Finally, let's just copy config to the class.
+
+    self.config = config
+
+.. _Writing-SAL-Scripts-Metadata:
+
+Metadata
+--------
+
+SAL Scripts provides a way to report information about itself to the system.
+These includes, amongst other things estimated duration, position on the sky and instrument configuration.
+
+The full set of metadata information is defined in the `Script.metadata`_ topic.
+
+.. _Script.metadata: https://ts-xml.lsst.io/sal_interfaces/Script.html#metadata
+
+Here we concentrate on filling up the most basic information, the estimated duration.
+This information is used by LOVE to provide feedback to the users about the execution progress.
+The metadata information is handled in the ``set_metadata`` method of the SAL Script:
```

Comment:
Is it worth adding that the scriptQueue will dump a script if it exceeds these estimates?
I think you want to add something about how accurate they need to be

Reply:
I don't think the ScriptQueue will do that. That may be a nice feature, though it would require some good thinking. 

We don't have a specific accuracy needed. At this point this is mainly used for information purposes.